nucypher/policy/maps.py

```diff
@@ -121,37 +122,47 @@ def __len__(self):
     def __bytes__(self):
         if self._payload is None:
             self._set_payload()
-        return self._payload
-
-    @classmethod
-    def splitter(cls):
-        return BytestringKwargifier(cls,
-                                    version=(bytes, cls._HEADER_SIZE),
-                                    public_signature=Signature,
-                                    hrac=(bytes, HRAC_LENGTH),
-                                    message_kit=(UmbralMessageKit, VariableLengthBytestring))
+        return self._BRAND + self._VERSION + self._payload
 
     @classmethod
-    def _check_version(cls, bytes_representation: bytes) -> None:
+    def get_splitter(cls, bytes_representation: bytes) -> None:
         """
         Takes a bytes representation of a treasure map and raises OldVersion
         error is the version is incompatible or ValueError if the header is malformed.
         """
-        header = bytes_representation[:cls._HEADER_SIZE]
-        if header != cls._HEADER:
-            if bytes_representation[:len(cls._PREFIX)] != cls._PREFIX:
-                # This either is not a treasure map, or predates versioning.
-                raise ValueError('Invalid treasure map header.')
-            try:
-                prefix, version = header.split(cls._DELIMITER)
-            except ValueError:
-                raise ValueError('Invalid treasure map header.')
-            raise cls.OldVersion(f'Treasure map is an old version ({version} but the latest version is {cls.VERSION_NUMBER}')
+        representation_metadata = TreasureMapSplitter.get_metadata(bytes_representation)
+
+        # header = bytes_representation[:cls._HEADER_SIZE]
+        brand_matches = representation_metadata['brand'] == cls._BRAND
+        version = representation_metadata['version']
+
+        if version <= cls.VERSION_NUMBER and brand_matches:
+            # This representation is compatible with a known stencil.
+            splitter = cls._splitters[version]
+        else:
+            # It's possible that this is a preversioned representation.
+            splitter = cls._splitters['unversioned']  # TODO: In this case, it's still a map from a previous version - how will we handle sin KFrags?
+        return splitter
+
+    _splitters = {}
+
+    def __new__(cls, *args, **kwargs):
```

Comment:
Can `_splitters` be defined as a class attribute? Looks like everything here is known at compile time.

Reply:
Oh yeah, that might now be the case.

Reply:
Actually, they do need the `cls` object.  Unless.... there's some better way... which there probably is....?

Reply:
Hm, I don't understand. What happens if you just create the `_splitters` class attribute with those two values?

Reply:
Ughh, what?

`NameError` for `cls` I suppose.

Reply:
That's right, I didn't realize `cls` itself is used here.